### PR TITLE
Add scriv package

### PR DESCRIPTION
**Pull request by sdebionne**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sdebionne** [comment]
@conda-forge/help-python, ready for review!

File: recipes/scriv/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{% set name = "scriv" %}
+{% set dunder_name = name.replace("-", "_") %}
```
Comment (ocefpaf) [inline recipes/scriv/meta.yaml:2]:
```suggestion
```

File: recipes/scriv/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{% set version = "1.5.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ dunder_name }}/{{ dunder_name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/scriv/meta.yaml:10]:
```suggestion
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipes/scriv/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python >=3.8
```
Comment (ocefpaf) [inline recipes/scriv/meta.yaml:25]:
```suggestion
    - python >=3.7
```

**Review by ocefpaf — COMMENTED** [review]
Please use grayskull to generate Python recipes. See https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#grayskull---recipe-generator-for-python-packages-on-pypi to make it easier to review.

File: recipes/scriv/meta.yaml (hunk 3 above)
Comment (sdebionne) [reply recipes/scriv/meta.yaml:25]: The upstream project requires 3.8 

https://github.com/nedbat/scriv/blob/16fe892e7c15f4a3a2bb104591389307c76ca2de/setup.cfg#L36C1-L36C17

Comment (ocefpaf) [reply recipes/scriv/meta.yaml:24]: In the future version. This one is 3.7 https://github.com/nedbat/scriv/blob/1.5.1/setup.cfg#L102

Comment (sdebionne) [reply recipes/scriv/meta.yaml:24]: Oh right but ITHO https://conda-forge.org/news/2022/08/24/dropping-python-37/

Comment (ocefpaf) [reply recipes/scriv/meta.yaml:24]: Sure, but that doesn't affect that this package will be, correctly installed on 3.7 if you follow the package metadata.

File: recipes/scriv/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+build:
+  entry_points:
+    - scriv = scriv.cli:cli
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python
```
Comment (ocefpaf) [inline recipes/scriv/meta.yaml:21]:
```suggestion
    - python >=3.7
```

File: recipes/scriv/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 30ae9ff8d144f8e0cf394c4e1d379542f1b3823767642955b54ec40dc00b32b6
+
+build:
+  entry_points:
+    - scriv = scriv.cli:cli
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/scriv/meta.yaml:17]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

**Pull request merged**